### PR TITLE
Refactor `Recall` tests

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Refactor tests, use pytest.mark.parametrize(), and improve test generation.

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/metrics/test_recall.py [hunk 1]
```diff
@@ -493,184 +231,63 @@ def to_numpy_multilabel(y):
...
+
+    def get_test_cases():
+
+        test_cases = [
+            (torch.randint(0, 2, size=(10, 5)), torch.randint(0, 2, size=(10, 5)), 1),
+            (torch.randint(0, 2, size=(10, 4)), torch.randint(0, 2, size=(10, 4)), 1),
+            # updated batches
+            (torch.randint(0, 2, size=(50, 5)), torch.randint(0, 2, size=(50, 5)), 16),
+            (torch.randint(0, 2, size=(50, 4)), torch.randint(0, 2, size=(50, 4)), 16),
+            (torch.randint(0, 2, size=(10, 5, 10)), torch.randint(0, 2, size=(10, 5, 10)), 1),
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_recall.py:272]: Please, add a comment here that it is another test case and not `# updated batches` anymore.

Comment (KickItLikeShika) [reply tests/ignite/metrics/test_recall.py:274]: Thanks! that's done now!

File: tests/ignite/metrics/test_recall.py [hunk 2]
```diff
@@ -493,184 +231,66 @@ def to_numpy_multilabel(y):
...
+        re1 = Recall(is_multilabel=True, average=True)
+        re2 = Recall(is_multilabel=True, average=False)
+        re1.update((y_pred, y))
+        re2.update((y_pred, y))
+        assert re1.compute() == pytest.approx(re2.compute().mean().item())
+
+    def get_test_cases():
+
+        test_cases = [
+            # 1st test cases
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_recall.py:267]: Well, this is not what we'd like to see :)
We had NCHW, NCL, NC use-cases. Can we update the comments here as for multiclass and binary ?
Thanks

**Comment by KickItLikeShika** [comment]
This link is not found https://www.aclweb.org/anthology/P02-1040.pdf
@gucifer can you have a look please?

**Comment by KickItLikeShika** [comment]
This link is not found https://www.aclweb.org/anthology/P02-1040.pdf
@sdesrozis please, can you have a look here? https://github.com/pytorch/ignite/pull/1906/checks?check_run_id=2270687894#step:6:466

**Comment by vfdev-5** [comment]
@KickItLikeShika I think we can replace the link as :
```diff
- https://www.aclweb.org/anthology/P02-1040.pdf
+ https://www.aclweb.org/anthology/P02-1040
```
in another PR

**Review by vfdev-5 — APPROVED** [review]
Thanks @KickItLikeShika , lgtm!

**Comment by gucifer** [comment]
> @KickItLikeShika I think we can replace the link as :
> 
> ```diff
> - https://www.aclweb.org/anthology/P02-1040.pdf
> + https://www.aclweb.org/anthology/P02-1040
> ```
> 
> in another PR

The pdf link is still not working from there as well. 
Let's add a working link like
https://dl.acm.org/doi/10.3115/1073083.1073135

**Comment by KickItLikeShika** [comment]
I agree with @gucifer

**Comment by KickItLikeShika** [comment]
@vfdev-5 the link works fine now, please restart the `Build docs` tests

**Pull request merged**